### PR TITLE
machine: Address some QEMU TODOs

### PR DESCRIPTION
**Pull request by jakecorrenti**
#### Does this PR introduce a user-facing change?

Addresses the following TODOs that were found in the podman 5 machine refactor code:
- We no longer care about the image provenance in `InspectInfo`, so remove the `Image` field
- Address whether `providerRm` failures should be a hard error
- Address whether a failure to remove machine files should be a hard error
- Implement QEMU-specific file removal
- Remove system connections first and remove machine config files last, this allows us to "see" the machine still in `podman machine list` if there was an error removing any prior files

```release-note
`ConfigPath` and `Image` are not longer provided to the user when they execute `podman machine inspect`. Users can also no longer use `{{ .ConfigPath }}` or `{{ .Image }}` when doing `podman machine inspect --format`
```

### CODE REVIEW

File: pkg/machine/qemu/machine.go [hunk 1]
```diff
@@ -236,7 +236,11 @@ func (q *QEMUStubber) Remove(mc *vmconfigs.MachineConfig) ([]string, func() erro
 	}
 
 	return qemuRmFiles, func() error {
-		return nil
+		if err := mc.QEMUHypervisor.QEMUPidPath.Delete(); err != nil {
+			return err
```
Comment (baude) [inline pkg/machine/qemu/machine.go:240]: if we fail here, we dont clean up the next one ... should above be a soft error (i.e. logrus) and the next one be the hard error (return err)

File: docs/source/markdown/podman-machine-inspect.1.md [hunk 2]
```diff
@@ -25,10 +25,8 @@ Print results with a Go template.
 
 | **Placeholder**     | **Description**                                                       |
 | ------------------- | --------------------------------------------------------------------- |
```
Comment (baude) [inline docs/source/markdown/podman-machine-inspect.1.md:27]: the changes here should probably make a release note

Comment (jakecorrenti) [reply docs/source/markdown/podman-machine-inspect.1.md:27]: Fixed.

**Comment by Luap99** [comment]
This needs to be targeted against main now.

File: cmd/podman/machine/rm.go [hunk 3]
```diff
@@ -120,14 +120,12 @@ func rm(_ *cobra.Command, args []string) error {
 	// All actual removal of files and vms should occur after this
 	//
 
-	// TODO Should this be a hard error?
 	if err := providerRm(); err != nil {
 		logrus.Errorf("failed to remove virtual machine from provider for %q", vmName)
 	}
 
-	// TODO Should this be a hard error?
```
Comment (Luap99) [inline cmd/podman/machine/rm.go:128]: Why are these removed? This almost certainly should be a hard error. Without this it will not exit >0 when it failed to remove the config.

File: pkg/machine/vmconfigs/machine.go [hunk 4]
```diff
@@ -185,28 +185,32 @@ func (mc *MachineConfig) Remove(saveIgnition, saveImage bool) ([]string, func()
 	}
 
 	mcRemove := func() error {
+		if err := connection.RemoveConnections(mc.Name, mc.Name+"-root"); err != nil {
+			return err
```
Comment (Luap99) [inline pkg/machine/vmconfigs/machine.go:189]: if we return right away here it will not delete the other things on error which is bad

This here should really use some of the multierror packages that we use and just store all errors and then as last step remove it, i.e. declare a `errs []error` slice at the top then append all error there instead of returning. Then at the function end return `errorhandling.JoinErrors(errs)`.

File: cmd/podman/machine/rm.go (hunk 3 above)
Comment (jakecorrenti) [reply cmd/podman/machine/rm.go:128]: You're right, thanks for pointing that out. Silly mistake. Should be fixed.

File: pkg/machine/vmconfigs/machine.go (hunk 4 above)
Comment (jakecorrenti) [reply pkg/machine/vmconfigs/machine.go]: I didn't know that existed, thanks Paul. Should be fixed as well.

File: cmd/podman/machine/rm.go [hunk 5]
```diff
@@ -120,14 +119,12 @@ func rm(_ *cobra.Command, args []string) error {
 	// All actual removal of files and vms should occur after this
 	//
 
-	// TODO Should this be a hard error?
 	if err := providerRm(); err != nil {
-		logrus.Errorf("failed to remove virtual machine from provider for %q", vmName)
+		return fmt.Errorf("failed to remove virtual machine from provider for %q: %v", vmName, err)
```
Comment (Luap99) [inline cmd/podman/machine/rm.go:123]: I think the one open question is do we want to call genericRm() even when this failed? I am not yet familiar with the differences between both but my gut tells me yes. So I would recommend to collect/log the errors like done elsewhere.
@baude WDYT?

Comment (jakecorrenti) [reply cmd/podman/machine/rm.go:123]: `providerRM` deals with provider specific files. So for QEMU we would remove the QMP Monitor pid file, for example. `genericRM` is where we delete the ignition file, config file, and image. I'm fine with logging the error and then doing a hard error on `genericRM`

Comment (rhatdan) [reply cmd/podman/machine/rm.go:123]: I would report the error either way but do both.  BTW Ignore ENOTEXIST errors on missing, (You can just logrus.Debug them.)

Comment (jakecorrenti) [reply cmd/podman/machine/rm.go:123]: `VMFile.Delete()` ignores ENOTEXIST when reporting the error so all good on that front

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM
One test still kicking up it's heals.

**Comment by jakecorrenti** [comment]
@TomSweeneyRedHat PTAL

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by mheon** [comment]
I have no objection.
/hold cancel
